### PR TITLE
Scream downstream merge 2024_10_09

### PR DESCRIPTION
**Pull request by jgfouca**
[BFB]

### CODE REVIEW

**Comment by jgfouca** [comment]
@mahf708 , feel free to push to my branch.

**Comment by jgfouca** [comment]
Merged to next.

File: components/eamxx/src/mct_coupling/scream_f2c_mod.F90 [hunk 1]
```diff
@@ -64,11 +72,19 @@ subroutine scream_setup_surface_coupling (import_field_names, import_cpl_indices
...
+#ifdef HAVE_MOAB
+    type(c_ptr),         intent(in) :: x2a_moab_ptr, a2x_moab_ptr
+#endif
     integer(kind=c_int), intent(in) :: num_cpl_imports, num_scream_imports, &
                                        num_cpl_exports, num_scream_exports
     integer(kind=c_int), intent(in) :: import_field_size, export_field_size
   end subroutine scream_setup_surface_coupling
 
+#if defined(SCREAM_SYSTEM_WORKAROUND) && (SCREAM_SYSTEM_WORKAROUND == 1)
+  subroutine scream_init_hip_atm () bind(c)
```
Comment (rljacob) [inline components/eamxx/src/mct_coupling/scream_f2c_mod.F90:84]: What is this for?

**Comment by jgfouca** [comment]
Merged fixes to next.

**Pull request merged**